app/main.py

```diff
@@ -1,34 +1,120 @@
+import dataclasses
+from typing import Callable
+from string import ascii_uppercase
+from collections import namedtuple
+
+
+class ValidationError(Exception):
+    pass
+
+
+@dataclasses.dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    row: int
+    column: int
+    is_alive: bool = True
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = {}
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        if start[0] != end[0]:
+            for row in range(start[0], end[0] + 1):
+                self.decks[(row, start[1])] = Deck(row, start[1])
+        else:
+            for column in range(start[1], end[1] + 1):
+                self.decks[(start[0], column)] = Deck(start[0], column)
```

Comment:
You can use loop inside loop
```suggestion
        for row in range(start[0], end[0] + 1):
            for column in range(start[1], end[1] + 1):
                self.decks[(start[0], column)] = Deck(start[0], column)
```

Reply:
Yes, but I need to create decks horizontally and vertically